### PR TITLE
Remove okhttp and use HttpClient from jdk

### PR DESCRIPTION
**Pull request by saberduck**
#2868 should be merged first!

### CODE REVIEW

**Comment by saberduck** [comment]
@vilchik-elena #2868 should be merged first! Also I suggest to merge with failing coverage - missing coverage is related to "impossible" exception handling

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java [hunk 1]
```diff
@@ -221,33 +217,36 @@ public void initLinter(List<Rule> rules, List<String> environments, List<String>
...
+      .uri(url(endpoint))
+      .timeout(Duration.ofSeconds(timeoutSeconds))
+      .header("Content-Type", "application/json")
+      .POST(HttpRequest.BodyPublishers.ofString(json))
       .build();
 
-    try (Response response = client.newCall(request).execute()) {
+    try {
+      var response = client.send(request, BodyHandlers.ofString());
       // in this case response.body() is never null (according to docs)
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java:247]: I am not sure we need this comment now

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java [hunk 2]
```diff
@@ -221,33 +217,36 @@ public void initLinter(List<Rule> rules, List<String> environments, List<String>
...
+  public AnalysisResponse analyzeCss(CssAnalysisRequest request) throws IOException, InterruptedException {
     String json = GSON.toJson(request);
     return response(request(json, "analyze-css"), request.filePath);
   }
 
-  private String request(String json, String endpoint) throws IOException {
-    Request request = new Request.Builder()
-      .url(url(endpoint))
-      .post(RequestBody.create(MediaType.get("application/json"), json))
+  private String request(String json, String endpoint) throws IOException, InterruptedException {
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java:237]: what is the value of propagating the `InterruptedException`? Can't we catch it here?

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java [hunk 3]
```diff
@@ -265,17 +264,16 @@ private static AnalysisResponse response(String result, String filePath) {
...
-      .build();
 
-    try (Response response = client.newCall(request).execute()) {
-      String body = response.body().string();
+    var request = HttpRequest.newBuilder(url("status")).GET().build();
+
+    try {
+      var response = client.send(request, BodyHandlers.ofString());
+      String body = response.body();
       // in this case response.body() is never null (according to docs)
```
Comment (vilchik-elena) [inline sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java:277]: same here

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/EslintBridgeIntegrationTest.java [hunk 4]
```diff
@@ -168,36 +165,28 @@ void start(Path dest) throws IOException {
...
         .build();
 
-      try (Response response = client.newCall(request).execute()) {
-        // in this case response.body() is never null (according to docs)
-        return response.body().string();
-      } catch (InterruptedIOException e) {
-        throw new IllegalStateException("eslint-bridge is unresponsive", e);
-      }
+      var response = client.send(request, HttpResponse.BodyHandlers.ofString());
+      // in this case response.body() is never null (according to docs)
```
Comment (vilchik-elena) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/EslintBridgeIntegrationTest.java:175]: remove comment

**Review by vilchik-elena — APPROVED** [review]
LGTM 
But I wonder if we can get rid of `InterruptedException` everywhere

File: sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java (hunk 2 above)
Comment (saberduck) [reply sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/eslint/EslintBridgeServerImpl.java]: I changed the code to rethrow as runtime exception

**Pull request merged**